### PR TITLE
add druid sink source

### PR DESCRIPTION
**Pull request by wenxuanguan**
## What changes were proposed in this pull request?

add druid sink source

## How was this patch tested?

kafka data sink to druid

### CODE REVIEW

File: emr-druid/src/main/scala/org/apache/spark/sql/druid/DruidSourceProvider.scala [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+      .map { k => k.drop(6).toString -> parameters(k) }
+      .toMap
+  }
+
+  override def sourceSchema(
+      sqlContext: SQLContext,
+      schema: Option[StructType],
+      providerName: String,
+      parameters: Map[String, String]): (String, StructType) = (shortName(), new StructType())
+
```
Comment (uncleGen) [inline emr-druid/src/main/scala/org/apache/spark/sql/druid/DruidSourceProvider.scala]: can not desc a table?

File: emr-druid/pom.xml [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+        <dependency>
+            <groupId>com.fasterxml.jackson.module</groupId>
+            <artifactId>jackson-module-scala_${scala.binary.version}</artifactId>
+            <version>${jackson.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
+            <version>15.0</version>
+        </dependency>
```
Comment (uncleGen) [inline emr-druid/pom.xml]: can not use guava version in root?

File: emr-druid/pom.xml [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>emr_2.11</artifactId>
+        <groupId>com.aliyun.emr</groupId>
+        <version>1.7.0-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
```
Comment (uncleGen) [inline emr-druid/pom.xml]: add a <name></name> filed

File: emr-druid/src/main/scala/org/apache/spark/sql/druid/DruidSourceProvider.scala (hunk 1 above)
Comment (wenxuanguan) [reply emr-druid/src/main/scala/org/apache/spark/sql/druid/DruidSourceProvider.scala]: added

File: emr-druid/pom.xml (hunk 2 above)
Comment (wenxuanguan) [reply emr-druid/pom.xml]: fixed.

File: emr-druid/pom.xml (hunk 3 above)
Comment (wenxuanguan) [reply emr-druid/pom.xml]: fixed

File: emr-druid/pom.xml [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+                        <includes>
+                            <include>org.skife.config:config-magic</include>
+                            <include>org.hibernate:hibernate-validator</include>
+                            <include>org.jboss.logging:jboss-logging</include>
+                            <include>com.fasterxml.jackson.datatype:jackson-datatype-guava</include>
+                            <include>com.fasterxml.jackson.dataformat:jackson-dataformat-smile</include>
+                            <include>com.google.inject.extensions:guice-multibindings</include>
+                            <include>org.scalaj:scalaj-time_2.11</include>
+                            <include>org.apache.curator:curator-x-discovery</include>
+                            <include>com.twitter:*</include>
```
[the remark this answers is not in the document]
Comment (wenxuanguan) [reply emr-druid/pom.xml]: deleted

File: emr-druid/src/main/scala/org/apache/spark/sql/aliyun/druid/DruidSource.scala [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+package org.apache.spark.sql.aliyun.druid
+
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.execution.streaming.{Offset, Source}
+import org.apache.spark.sql.types.StructType
+import org.apache.spark.sql.{DataFrame, SQLContext}
+import org.apache.spark.{Partition, SparkContext, TaskContext}
+
+class DruidSource(
```
[the remark this answers is not in the document]
Comment (wenxuanguan) [reply emr-druid/src/main/scala/org/apache/spark/sql/aliyun/druid/DruidSource.scala]: not supported

**Pull request merged**